src/main/java/org/opensearch/flowframework/transport/ProvisionWorkflowTransportAction.java

```diff
@@ -156,13 +159,42 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
     private void executeWorkflowAsync(String workflowId, List<ProcessNode> workflowSequence) {
         // TODO : Update Action listener type to State index Request
         ActionListener<String> provisionWorkflowListener = ActionListener.wrap(response -> {
-            logger.info("Provisioning completed successuflly for workflow {}", workflowId);
+            logger.info("Provisioning completed successfully for workflow {}", workflowId);
+            flowFrameworkIndicesHandler.updateFlowFrameworkSystemIndexDoc(
+                WORKFLOW_STATE_INDEX,
+                workflowId,
+                ImmutableMap.of(
+                    STATE_FIELD,
+                    State.COMPLETED,
```

Comment:
Are we changing this state later, if failed?

Reply:
yes below on line 190